### DISCUSSION TITLE
How to install or update Winget by using command line?

### DISCUSSION THREAD

**Discussion by Tataz2** [microsoft/winget-cli#4406] [Q&A]
Hello

I need to install software by Winget on freshly installed Windows 10 computer but Winget does not work and I cannot update it.

```
C:\Windows\system32>winget -v
v1.5.2201
```

Cannot install any packages. I get an error.
`Failed when opening source(s); try the 'source reset' command if the problem persists.`

`winget source reset` does not help.

If I install winget by command
`Add-AppxPackage -Path https://cdn.winget.microsoft.com/cache/source.msix`

The version is still the same

```
C:\Windows\system32>winget -v
v1.5.2201
```

but now I get an error:
```
Failed in attempting to update the source: winget
Failed when searching source: msstore
An unexpected error occurred while executing the command:
0x8a15005e : The server certificate did not match any of the expected values.
```
```
PS C:\Windows\system32> winget source update
Updating all sources...
Updating source: msstore...
Done
Updating source: winget...
Cancelled
```

I have tried to update Winget by installing Microsoft.DesktopAppInstaller_8wekyb3d8bbwe.txt.msixbundle
`Add-AppxPackage -Path "https://aka.ms/getwinget" `
but even it's msixbundle it doesn't work.
```
Windows cannot install package Microsoft.DesktopAppInstaller_1.22.11132.0_x64__8wekyb3d8bbwe because this package depen
ds on a framework that could not be found. Provide the framework "Microsoft.UI.Xaml.2.8" published by "CN=Microsoft Cor
poration, O=Microsoft Corporation, L=Redmond, S=Washington, C=US", with neutral or x64 processor architecture and minim
um version 8.2310.30001.0, along with this package to install. The frameworks with name "Microsoft.UI.Xaml.2.8" current
ly installed are: {Microsoft.UI.Xaml
Windows cannot install package Microsoft.DesktopAppInstaller_1.22.11132.0_x64__8wekyb3d8bbwe because this package depen
ds on a framework that could not be found. Provide the framework "Microsoft.UI.Xaml.2.8" published by "CN=Microsoft Cor
poration, O=Microsoft Corporation, L=Redmond, S=Washington, C=US", with neutral or x64 processor architecture and minim
um version 8.2310.30001.0, along with this package to install. The frameworks with name "Microsoft.UI.Xaml.2.8" current
ly installed are: {Microsoft.UI.Xaml.2.8_8.2306.22001.0_x64__8wekyb3d8bbwe Microsoft.UI.Xaml.2.8_8.2306.22001.0_x86__8w
ekyb3d8bbwe}
NOTE: For additional information, look for [ActivityId] 1c44a6e6-9617-0000-d3bf-441c1796da01 in the Event Log or use th
e command line Get-AppPackageLog -ActivityID 1c44a6e6-9617-0000-d3bf-441c1796da01
```

**Comment by AdmixxTTP** [comment]
Hey,
maybe you can try this in PowerShell (as Admin), it has helped for me...

Add-AppxPackage -Path 'https://github.com/microsoft/winget-cli/releases/download/v1.7.11132/Microsoft.DesktopAppInstaller_8wekyb3d8bbwe.msixbundle'

In some cases, you may need to add -ForceApplicationShutdown to the end of the command in order to update the package successfully

Br

**Reply by Tataz2** [reply]
> Hey, maybe you can try this in PowerShell (as Admin), it has helped for me...
> 
> Add-AppxPackage -Path 'https://github.com/microsoft/winget-cli/releases/download/v1.7.11132/Microsoft.DesktopAppInstaller_8wekyb3d8bbwe.msixbundle'
> 
> In some cases, you may need to add -ForceApplicationShutdown to the end of the command in order to update the package successfully
> 
> Br

The source "https://github.com/microsoft/winget-cli/releases/download/v1.7.11132/Microsoft.DesktopAppInstaller_8wekyb3d8bbwe.msixbundle" is is same as the "https://aka.ms/getwinget"  so this won't help. 

It would be nice if Microsoft.DesktopAppInstaller_8wekyb3d8bbwe.msixbundle included all dependencies. Or if there was a general install url like https://aka.ms/getwinget for the latest Microsoft.UI.Xaml.2.8.

**Comment by Tataz2** [comment]
This works at the moment but the link for Microsoft.UI.Xaml.2.8 might get old.
```
IF ([System.Version](Get-AppxPackage Microsoft.DesktopAppInstaller).Version -lt [System.Version]'1.22.1000' ) {
Add-AppxPackage -Path "https://aka.ms/Microsoft.VCLibs.x64.14.00.Desktop.appx"
Add-AppxPackage -Path "https://github.com/microsoft/microsoft-ui-xaml/releases/download/v2.8.6/Microsoft.UI.Xaml.2.8.x64.appx"
Add-AppxPackage -Path "https://aka.ms/getwinget"
}
```

**Comment by mdanish-kh** [comment]
If you are on WinGet 1.6+ (so to have dependencies support), you can use `winget upgrade Microsoft.AppInstaller` or `winget upgrade winget` to upgrade winget.

If you are doing this on a fresh machine that may have an older version of WinGet, I would suggest using this sandbox script:

https://learn.microsoft.com/en-us/windows/package-manager/winget/#install-winget-on-windows-sandbox

**Reply by Tataz2** [reply]
> If you are on WinGet 1.6+ (so to have dependencies support), you can use `winget upgrade Microsoft.AppInstaller` or `winget upgrade winget` to upgrade winget.
> 
> If you are doing this on a fresh machine that may have an older version of WinGet, I would suggest using this sandbox script:
> 
> https://learn.microsoft.com/en-us/windows/package-manager/winget/#install-winget-on-windows-sandbox

Yes fresh machine and first login to dekstop. Windows 10 2004.
